docs/content/documentation/systems/Problem/contact/ReferenceResidualProblem.md

```diff
@@ -1,8 +1,30 @@
-<!-- MOOSE Documentation Stub: Remove this when content is added. -->
-
 # ReferenceResidualProblem
 !syntax description /Problem/ReferenceResidualProblem
 
+## Description
+
```

Comment:
Replace everything under Description with this:

By default, MOOSE checks convergence using relative and absolute criteria. Once the residual drops below either an absolute tolerance, or the residual divided by the initial residual for the current time step drops below a relative tolerance, the solution is considered converged. This works well for many problems, but in cases where the solution changes minimally between time steps, the initial residual can be very small, making the relative convergence check much too stringent.

The `ReferenceResidualProblem` checks for convergence by comparing the residual to a different reference quantity (instead of the initial residual). The the user supplies a reference quantity in the form of a set of `AuxVariables` that contain physically meaningful quantities that can be used in a relative convergence check.  Because the solution variables can have significantly different scaling, the convergence check performed in `ReferenceResidualProblem` checks convergence of the solution variables individually. When the $$l^2$$ norm of the residual for each solution variable is less than either the relative tolerance times the $$l^2$$ norm of the corresponding reference variable or the absolute tolerance, the solution is considered converged.

As mentioned above, use of this procedure requires that the user provide physically meaningful reference quantities. The vector of the reaction loads (in the case of mechanics) or integrated fluxes (in the case of diffusion problems) is typically suitable for this purpose, as it provides a measure of the loading applied to the system. To compute these, an `AuxVariable` must be set up corresponding to each solution variable, and the  `save_in` option is used in each `Kernel` to assemble the residual into that vector. When the solution is converged, that `AuxVariable` will contain values that are close to zero everywhere except for where boundary conditions are applied.

Since relative convergence is computed differently with this approach, the nonlinear relative tolerance required to achieve the same error is typically different than with the default approach in MOOSE, and the differences will vary by the problem. The code user must evaluate the behavior of their model to ensure that appropriate tolerances are being used.